test/devicecontroller/DefaultDeviceController.test.ts

```diff
@@ -2106,4 +2106,38 @@ describe('DefaultDeviceController', () => {
       deviceController.chooseAudioInputDevice(stringDeviceId);
     });
   });
+
+  describe('getIntrinsicDeviceId', () => {
+    it('Return undefined if the device is undefined', () => {
+      expect(DefaultDeviceController.getIntrinsicDeviceId(undefined)).to.be.undefined;
+    });
+
+    it('Return null if the device is null', () => {
+      expect(DefaultDeviceController.getIntrinsicDeviceId(null)).to.be.null;
+    });
+
+    it('Return string if the device is of type string', () => {
+      expect(DefaultDeviceController.getIntrinsicDeviceId('')).to.equal('');
+    });
```

Comment:
Also test `'default'` and a real device ID value.

Reply:
Updated.